SAK-30075 - Web service for finding and fixing bad attachments in Sam…

danielmerino opened the pull request:
…igo pools

ern commented on webservices/cxf/src/java/org/sakaiproject/webservices/TestsAndQuizzes.java:
```diff
@@ -330,4 +334,92 @@ private String findAssessmentTemplateId(String title) {
+				while (qpif.hasNext())
+				{
+					QuestionPoolFacade qp = (QuestionPoolFacade) qpif.next();
+					resultado.append(this.poolAttachmentReport(sessionId, user, qp.getQuestionPoolId(), contextToReplace));
```
One public web service calling another is probably not ideal! It should just call the service appropriately.

I would even suggest just combining these into one service.

danielmerino replied:
Do you mean calling poolAttachmentReport from userPoolsAttachmentReport? It was just for not repeating code. The first one looks for attachments in one single pool. The second one search in all pools related to user, so is calling the first for each pool. How do you suggest combining them?

ern replied:
Forexample of the poolId was null then do all pools for that user otherwise you would do just that pool for that user.

danielmerino replied:
It's a good idea but it seems that CXF does not allow passing null values as service parameters. What about value 0 instead of null for poolId?

ern replied:
yeah in rpc literal null values are not allowed but an empty string should be fine

ern commented:
There is an easy to use test framework for soap web services it would be nice if tests were add for new web services.

ern commented:
I understand this web service is used to fix a broken attachment condition within question pools but has the root issue been resolved within the question pool logic?

danielmerino commented:
This issue is being addressed at SAM-2257.

El 16 de noviembre de 2015 15:45:05 CET, Earle Nietzel <notifications@github.com> escribió:
>I understand this web service is used to fix a broken attachment
>condition within question pools but has the root issue been resolved
>within the question pool logic?
>
>---
>Reply to this email directly or view it on GitHub:
>https://github.com/sakaiproject/sakai/pull/1272#issuecomment-157054834

-- 
Sent from my Android device with K-9 Mail. Please excuse my brevity.

danielmerino commented:
I have refactored all the Samigo code and I have put it inside QuestionPoolServiceAPI bean, so I can inject it into webservices and include a mock.

I have also changed it to keep one single method with String parameters, in order that empty strings are treated like null values.

ern commented on webservices/cxf/pom.xml:
```diff
@@ -179,6 +179,16 @@
+            <groupId>OKI</groupId>
+            <artifactId>OkiSID</artifactId>
+            <version>rc6.1</version>
+        </dependency>
```
Are these dependencies still needed? If so can you fix the formatting!!!

danielmerino replied:
Not needed for the WS but are needed for the test. Format fixed.

juanjmerono replied:
Consider to include <scope>test</scope> for both dependencies in that case, and also remove version from OkiOSID artifact to inherit master pom version.

danielmerino replied:
Done. Thanks!

ern commented on webservices/cxf/src/test/java/org/sakaiproject/webservices/MockingAbstractWebService.java:
```diff
@@ -44,6 +44,8 @@
 import org.sakaiproject.tool.api.ToolManager;
 import org.sakaiproject.tool.assessment.samlite.api.SamLiteService;
 import org.sakaiproject.user.api.UserDirectoryService;
+//import org.sakaiproject.tool.assessment.services.QuestionPoolService;
```
Remove this line?

ern commented on webservices/cxf/src/test/java/org/sakaiproject/webservices/TestsAndQuizzesPoolAttachmentReportTest.java:
```diff
@@ -0,0 +1,89 @@
+			
+		}
+		
+		//QuestionPoolService mockQuestionPoolService = mock(QuestionPoolService.class);
```
Remove this line?
